src/Layers/DynamicMapLayer.js

```diff
@@ -42,6 +42,12 @@ export var DynamicMapLayer = RasterLayer.extend({
   },
 
   setLayers: function (layers) {
+    if (!layers) { layers = []; }
+    if (layers.length === 0 && this.options.layers.length) {
+      this.onRemove(this._map);
+    } else if (layers.length && this.options.layers.length === 0) {
+      this.onAdd(this._map);
+    }
     this.options.layers = layers;
     this._update();
```

Comment:
now the ._update() call throws `Uncaught TypeError: Cannot read property 'removeLayer' of null` in the following situation.

```js
.setLayers([1]);
// wait
.setLayers([]);
```
hopefully its just an order of operations thing.